test/common/common/hash_test.cc

```diff
@@ -18,4 +18,24 @@ TEST(Hash, djb2CaseInsensitiveHash) {
   EXPECT_EQ(7195212308U, HashUtil::djb2CaseInsensitiveHash("lyft"));
   EXPECT_EQ(5381U, HashUtil::djb2CaseInsensitiveHash(""));
 }
+
+TEST(Hash, murmurHash2_64) {
+  EXPECT_EQ(9631199822919835226U, MurmurHash::murmurHash2_64("foo"));
+  EXPECT_EQ(11474628671133349555U, MurmurHash::murmurHash2_64("bar"));
+  EXPECT_EQ(16306510975912980159U, MurmurHash::murmurHash2_64("foo\nbar"));
+  EXPECT_EQ(12847078931730529320U, MurmurHash::murmurHash2_64("lyft"));
+  EXPECT_EQ(6142509188972423790U, MurmurHash::murmurHash2_64(""));
+}
+
+#ifndef __APPLE__
```

Comment:
Per pervious discussion, this fails because of an incompatible `std::hash` implementation in libc++, so this issue isn't macOS specific.

Could you replace this check with `#ifndef _LIBCPP_VERSION` to detect libc++ (same for the existing check in `ring_hash_lb_test.cc`)? Thanks!

Reply:
That would cause more issues down the line, also since the spec change in C++ 14:
```
Hash functions are only required to produce the same result for the same input within a single execution of a program; this allows salted hashes that prevent collision DoS attacks. | (since C++14)
```
I'll check for known compatible libstdc++ versions instead.

Reply:
@HenryYYang this code is all getting deleted pretty much once you merge this PR, so I wouldn't put too much effort into it.